nextcloud/server
# Fix state propragation of the backup codes provider

Starting with Nextcloud 14, the server knows the enabled/disabled
state of 2fa providers. While it will query that information if it's
unknown (on first use), it won't notice any changes. Thus, providers
have to propagate that information themselves.

Ref https://github.com/nextcloud/twofactor_totp/pull/263
Ref https://github.com/nextcloud/twofactor_u2f/pull/210
Ref https://github.com/orgs/nextcloud/projects/4#card-10826964

---

## Test steps

1. Switch to master
2. Make sure you don't have backup codes generated, if you have, create a new account.
3. Log in and create backup codes
4. Configure another 2FA provider
5. Log out
6. Log in

### Expected behavior

See regular 2FA provider *and* backup codes option.

### Actual behavior

See regular 2FA provider, but miss backup codes option.

To see the expected behavior on this branch, you'll have to re-generate the backup codes.
